### ISSUE TITLE
`not-empty-atom` lint incorrectly flags atoms with `λ` marker having a non-empty `@base` attribute

### ISSUE THREAD

**Issue by morphqdd** [objectionary/lints#861]
**What happens.** The `not-empty-atom` lint (rule `src/main/resources/org/eolang/lints/atoms/not-empty-atom.xsl`) flags every atom whose lambda marker `λ` carries a non-empty `base` attribute. After the recent restoration of the `/name` return-type syntax in `objectionary/eo` (#4707), atoms like `[] > number` with `[x] > times /number` produce XMIR where the `λ` marker now has `@base` set to the return type, e.g.:

```xml
<o name="number">
  <o base="number" name="λ"/>
</o>
```

The XSL selects:

```xpath
//o[eo:atom(.) and o[@base and @base!='∅' and not(@base='ξ' and @name='xi🌵')]]
```

This matches the atom because the `λ` child has `@base` and the only excluded case is `@base='ξ' and @name='xi🌵'`. As a result, the lint reports a false positive for any atom with parameters or a typed return.

The downstream effect: `objectionary/eo` had to add `+unlint not-empty-atom` to **22 `.eo` files** in `eo-runtime` (`bytes.eo`, `i16.eo`, `i32.eo`, `i64.eo`, `number.eo`, `try.eo`, `error.eo`, `malloc.eo`, `fs/dir.eo`, `fs/file.eo`, `ms/sqrt.eo`, `ms/acos.eo`, `ms/asin.eo`, `ms/pow.eo`, `ms/ln.eo`, `ms/angle.eo`, `sm/win32.eo`, `sm/os.eo`, `sm/posix.eo`, `tt/sprintf.eo`, `tt/sscanf.eo`, `tt/regex.eo`) just to keep its build green. The puzzle [`#4707-ff431d75`](https://github.com/objectionary/eo/issues/5091) tracks the cleanup on the `eo` side once this lint is fixed.

**What should happen.** The `λ` marker is part of the atom syntax, not an inner object. The XPath in `not-empty-atom.xsl` should also exclude children whose `@name='λ'` (analogously to the existing `@base='ξ' and @name='xi🌵'` exclusion). For example:

```xpath
//o[eo:atom(.) and o[@base and @base!='∅' and not(@name='λ') and not(@base='ξ' and @name='xi🌵')]]
```

with a regression test pack that asserts zero defects on a document like:

```xml
<object author="tests">
  <o name="number">
    <o base="number" name="λ"/>
  </o>
</object>
```

After the fix, `objectionary/eo` can bump the `lints` dependency in `eo-maven-plugin/pom.xml` (currently `0.0.60`) and drop the `+unlint not-empty-atom` lines from all 22 `.eo` files.

This bug is documented in the puzzle text in [`bytes.eo`](https://github.com/objectionary/eo/blob/master/eo-runtime/src/main/eo/bytes.eo) but no upstream issue exists in `objectionary/lints` to track the fix.

**Comment by morphqdd** [comment]
@yegor256, please take a look when you have a moment.

**Comment by morphqdd** [comment]
@yegor256, gentle reminder — could you take a look when time permits?

**Comment by 0crat** [comment]
@morphqdd Thanks for reporting a new bug! According to our work policy, you've earned **+12 points** as your bonus for this accepted bug report. By reporting bugs, you help our project improve its quality and earn rewards in the process. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it for more point bonuses. Your running score is [+224](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too.
